storage/replica_test.go

```diff
@@ -2675,6 +2683,137 @@ func TestRangeDanglingMetaIntent(t *testing.T) {
 	}
 }
 
+// TestRangeLookupUseReverseScan verifies the correctness of the results which are retrived
+// from RangeLookup by using ReverseScan.
+func TestRangeLookupUseReverseScan(t *testing.T) {
+	defer leaktest.AfterTest(t)
+	tc := testContext{}
+	tc.Start(t)
+	defer tc.Stop()
+
+	// Test ranges: ["a","c"), ["c","f"), ["f","h") and ["h","y").
+	testRanges := []proto.RangeDescriptor{
+		{RangeID: 2, StartKey: proto.Key("a"), EndKey: proto.Key("c")},
+		{RangeID: 3, StartKey: proto.Key("c"), EndKey: proto.Key("f")},
+		{RangeID: 4, StartKey: proto.Key("f"), EndKey: proto.Key("h")},
+		{RangeID: 5, StartKey: proto.Key("h"), EndKey: proto.Key("y")},
+	}
+	// The range ["f","h") has dangling intent in meta2.
+	withIntentRangeIndex := 2
+
+	testCases1 := []struct {
+		key      string
+		expected proto.RangeDescriptor
+	}{
+		// testRanges[0|1] exists, use the key in the midlle or end of
+		// the range will get itself.
+		{key: "b", expected: testRanges[0]},
+		{key: "c", expected: testRanges[0]},
+		{key: "d", expected: testRanges[1]},
+		{key: "f", expected: testRanges[1]},
+		// testRanges[2] does not exist or has intent. If testRanges[2]
+		// has intent, scan inconsistently will read an old value. But
+		// it only has one version, nil will be retured by MVCCScan. The
+		// end result (testRanges[1]) will be returned by MVCCReverseScan.
+		{key: "g", expected: testRanges[1]},
+		{key: "h", expected: testRanges[1]},
+		// testRanges[3] exists, use the key in the midlle or end of
```

Comment:
just move this case up above testRanges[2], then you can generalize the above comment to testRanges[0|1|3].